src/realm/object-store/impl/deep_change_checker.cpp

```diff
@@ -108,6 +108,25 @@ DeepChangeChecker::DeepChangeChecker(TransactionChangeInfo const& info, Table co
 {
 }
 
+template <typename T>
+bool DeepChangeChecker::do_check_for_collection_of_mixed(T* coll, size_t depth)
+{
+    TableRef cached_linked_table;
+    return std::any_of(coll->begin(), coll->end(), [&, this](auto value) {
+        if (value.is_type(type_TypedLink)) {
+            auto link = value.get_link();
+            if (!link.is_unresolved()) {
+                if (!cached_linked_table || cached_linked_table->get_key() != link.get_table_key()) {
+                    cached_linked_table = coll->get_table()->get_parent_group()->get_table(link.get_table_key());
+                    REALM_ASSERT_EX(cached_linked_table, link.get_table_key().value);
+                }
+                return this->check_row(*cached_linked_table, link.get_obj_key().value, depth + 1);
+            }
+        }
+        return false;
```

Comment:
It looks like this part of the function is shared between list/set/dictionary and could be extracted to a templated function?